### PR TITLE
Added AWS CodeBuild to CICD doc /astronomer/issues/issues/1884

### PR DESCRIPTION
**Pull request by ben-astro**
Added AWS CodeBuild to CICD doc and changed headers of example code/scripts so they are linkable (only H2 headers have built-in links)

### CODE REVIEW

**Comment by ben-astro** [comment]
FYI: Build-spec for this from AWS docs: https://docs.aws.amazon.com/codebuild/latest/userguide/build-spec-ref.html

**Review by paolaperaza — APPROVED** [review]
Looks good @ben-astro ! Went in there and adjusted spacing for a few lines. Merging!

**Review by paolaperaza — APPROVED** [review]

File: cloud/stable/03_deploy/06_ci-cd.md [hunk 1]
```diff
@@ -318,13 +317,36 @@ astro_deploy:
...
+    runtime-versions:
+      python: latest
+​
+  pre_build:
+    commands:
+      - echo Logging in to dockerhub ...
+      - docker login "registry.$BASE_DOMAIN" -u _ -p "$API_KEY_SECRET"
+      - export GIT_VERSION="$(git rev-parse --short HEAD)"
+      - echo "GIT_VERSION = $GIT_VERSION"
+      - pip install -r requirements.txt
```
Comment (vparekh94) [inline cloud/stable/03_deploy/06_ci-cd.md:335]: We don't need this line in our example.

File: cloud/stable/03_deploy/06_ci-cd.md [hunk 2]
```diff
@@ -318,13 +317,36 @@ astro_deploy:
...
+```yaml
+version: 0.2
+phases:
+  install:
+    runtime-versions:
+      python: latest
+​
+  pre_build:
+    commands:
+      - echo Logging in to dockerhub ...
```
Comment (vparekh94) [inline cloud/stable/03_deploy/06_ci-cd.md:331]: We're not logging into Dockerhub here; we'd be logging into Astronomer

File: cloud/stable/03_deploy/06_ci-cd.md [hunk 3]
```diff
@@ -318,13 +317,36 @@ astro_deploy:
...
+      - docker login "registry.$BASE_DOMAIN" -u _ -p "$API_KEY_SECRET"
+      - export GIT_VERSION="$(git rev-parse --short HEAD)"
```
Comment (vparekh94) [inline cloud/stable/03_deploy/06_ci-cd.md:333]: We don't _need_ either of these git lines either. I'd leave them out of our example

File: cloud/stable/03_deploy/06_ci-cd.md (hunk 1 above)
Comment (paolaperaza) [reply cloud/stable/03_deploy/06_ci-cd.md:335]: @vparekh94 Is there a reason the customer who sent us this had these lines in there? Presume they got some value out of it, but maybe not anything that needs to be generalizable? We can alternatively leave them in there and add a note that says these are optional

Will defer to you

**Pull request merged**